### ISSUE TITLE
Create multi-dataset overlay visualization dashboard

### ISSUE THREAD

**Issue by NCAsterism** [NCAsterism/places2go#21]
## Overview
Create a comprehensive HTML dashboard that overlays all datasets to reveal correlations and insights.

## Objectives
- Interactive map with destinations, flight routes, and weather overlays
- Unified timeline showing flights and weather together
- Cost vs weather/location analysis
- Best value finder (low cost + good weather + cheap flights)
- Destination comparison tool
- Export insights as report

## Data Sources
- All 4 CSV files via DataLoader
- Merged view: `loader.load_all(data_source='demo1')`
- Aggregates: `loader.get_aggregates(data_source='demo1')`

## Technical Requirements
- Use Plotly Dash or single-page HTML with multiple subplots
- Generate standalone HTML file in `.build/visualizations/`
- Consistent color scheme across all visualizations
- Cross-filtering (select on one chart filters others)
- Responsive layout with tabs or sections
- Performance optimized for 42+ records

## Visualizations
1. **Master Map**: 
   - Destinations plotted with size = cost, color = avg temp
   - Flight routes as curved lines (thickness = price)
   - Weather overlay with icons
   - Click destination to filter all charts

2. **Timeline View**:
   - X-axis: dates (Oct 5-17)
   - Top: Flight prices line chart
   - Bottom: Weather conditions bars
   - Aligned by date for correlation

3. **Value Finder**:
   - Scatter plot: X=cost, Y=avg_temp, size=flight_price
   - Quadrants: "Best Value" (low cost, good weather, cheap flights)
   - Interactive selection

4. **Destination Comparison**:
   - Radar chart comparing 2-3 destinations
   - Metrics: cost, weather, flight price, sunshine, etc.
   - Normalized 0-100 scale

5. **Insights Panel**:
   - Best value destination
   - Cheapest flights by day
   - Best weather days
   - Cost savings summary

## Acceptance Criteria
- [ ] HTML file with integrated multi-dataset view
- [ ] Map shows all destinations with overlays
- [ ] Timeline aligns flights and weather
- [ ] Value finder quadrant chart working
- [ ] Destination comparison radar chart
- [ ] Cross-filtering between charts
- [ ] Insights panel with calculated recommendations
- [ ] All data sources displayed
- [ ] Color consistency across charts
- [ ] File saved to `.build/visualizations/dashboard_overlay.html`
- [ ] Documentation in visualization README
- [ ] Performance: loads < 2 seconds

## Dependencies
- Requires issues #1-4 completed first
- Build on existing visualization patterns

## Future Enhancement
- Add real-time API data integration
- Machine learning predictions
- User preferences input
- Save/share custom views

## Estimated Effort
8-10 hours (after individual visualizations complete)

**Comment by NCAsterism** [comment]
@copilot This is the final Phase 3C deliverable - the multi-dataset overlay dashboard that integrates all 4 individual visualizations (weather, flights, costs, destinations) into a unified interface.

Please implement the overlay dashboard with:
- Master map with all overlays
- Timeline view aligning flights and weather
- Value finder scatter plot
- Destination comparison radar chart
- Insights panel with recommendations
- Cross-filtering between charts

See \docs/project/PHASE3C_PLAN.md\ for detailed architecture and implementation guidance.

Use the existing visualization modules from Phase 3B:
- \scripts/visualizations/weather_forecast.py\
- \scripts/visualizations/flight_prices.py\
- \scripts/visualizations/cost_comparison.py\
- \scripts/visualizations/destinations_map.py\

And the DataLoader for data access:
\\\python
loader = DataLoader()
destinations = loader.load_destinations()
costs = loader.load_costs(data_source='demo1')
flights = loader.load_flights(data_source='demo1')
weather = loader.load_weather(data_source='demo1', forecast_only=True)
\\\

Please create \scripts/visualizations/overlay_dashboard.py\ and generate \.build/visualizations/overlay_dashboard.html\.
